src/ctest.ts

```diff
@@ -602,14 +604,40 @@ export class CTestDriver implements vscode.Disposable {
         return uniqueTests;
     }
 
-    private async runTestHandler(request: vscode.TestRunRequest, cancellation: vscode.CancellationToken, useCMakePresets: boolean) {
+    // Returns false if test preset wasn't selected already (changing test preset triggers test explorer refresh)
+    private async checkTestPreset(tests: vscode.TestItem[]): Promise<boolean> {
+        let presetMayChange = false;
+        for (const test of tests) {
+            if (test.id === testPresetRequired) {
+                const folder = test.parent ? test.parent.id : test.id;
+                const project = await this.projectController?.getProjectForFolder(folder);
+                if (!project) {
+                    log.error(localize('no.project.found', 'No project found for folder {0}', folder));
+                    return false;
+                }
+                await vscode.commands.executeCommand('cmake.selectTestPreset', project.workspaceFolder);
+                presetMayChange = true;
```

Comment:
After this, you probably want to break out of the loop or it will run for all the tests.

Reply:
I'd like to ask the user to select test presets for all folders that that user want to run tests. Do you think picking one each time a better option?

Reply:
It looks like you select presets for every test in every folder.  I think you need to track whether you've already set the preset or not so that you don't prompt the user N times.